src/main/java/net/minestom/server/network/packet/server/play/WindowItemsPacket.java

```diff
@@ -32,4 +39,22 @@ public void write(@NotNull BinaryWriter writer) {
     public int getId() {
         return ServerPacketIdentifier.WINDOW_ITEMS;
     }
+
+    @Override
+    public @NotNull Collection<Component> components() {
+        final var list = new ArrayList<>(this.items);
+        list.add(this.carriedItem);
+
+        return list.stream().map(ItemStack::getDisplayName).filter(Objects::nonNull).collect(Collectors.toList());
```

Comment:
There is absolutely no reason to use `collect(Collectors.toList())` over `toList()` from JDK 17